### PR TITLE
fix(shared): Markdown links always open in a new tab

### PR DESCRIPTION
**Pull request by HjorturJ**
# All markdown links should open in a new tab by default

https://app.asana.com/0/1207402031871459/1208255253643787/f

## What

Making all markdown links open in a new tab by default

## Why

Because it makes the most sense and was requested. Still possible to overwrite and have links open in the same tab

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced the Markdown component to support optional rendering options, including the ability to open links in a new tab.
  
- **Bug Fixes**
  - Improved the flexibility of Markdown rendering with updated options handling.

- **Documentation**
  - Updated interface and method signatures to reflect new options for the Markdown component.

### CODE REVIEW

File: libs/shared/components/src/Markdown/Markdown.tsx [hunk 1]
```diff
@@ -1,11 +1,24 @@
 import { compiler } from 'markdown-to-jsx'
-import { markdownOptions } from './markdownOptions'
+import { getMarkdownOptions } from './markdownOptions'
 
 interface Props {
   children: string
+  conditionals?: OptionalConditionalOverrides
```
Comment (Toti91) [inline libs/shared/components/src/Markdown/Markdown.tsx:6]: maybe rename to "options", conditionals doesn't really describe what we are doing here I feel like

File: libs/shared/components/src/Markdown/Markdown.tsx [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 }
 
-export const Markdown = ({ children }: Props) =>
-  compiler(children, markdownOptions)
+export interface OptionalConditionalOverrides {
+  openLinksInNewTab?: boolean
+  hehe?: number
```
Comment (Toti91) [inline libs/shared/components/src/Markdown/Markdown.tsx:11]: hehe

**Review by Toti91 — CHANGES_REQUESTED** [review]
Now I don't know if this Markdown component is used outside of the application-system. But maybe we should have it default to false for opening links in new tab, to keep the current behavior, and then just update our usage  of this component in the application-system to override the default

File: libs/shared/components/src/Markdown/Markdown.tsx (hunk 2 above)
Comment (HjorturJ) [reply libs/shared/components/src/Markdown/Markdown.tsx]: image

File: libs/shared/components/src/Markdown/Markdown.tsx (hunk 1 above)
Comment (HjorturJ) [reply libs/shared/components/src/Markdown/Markdown.tsx]: fair, fixed

**Review by Toti91 — APPROVED** [review]
LGTM

**Review by magnearun — APPROVED** [review]

**Pull request merged**